cheatsheet.md

```diff
@@ -0,0 +1,162 @@
+    if (m_itemNo < SHIELD && m_itemNo == PENDANT) {
+        if (mAcch.m_flags & dBgS_Acch::GROUND_LANDING) {
+            // stuff
+        }
+    } else if (mAcch.m_flags & dBgS_Acch::GROUND_LANDING) {
+        // stuff 1
+    }
+}
+// Should be
+switch (m_itemNo) {
+case SHIELD:
+    break;
+case PENDANT:
+    if (mAcch.ChkGroundLanding()) {
+        // stuff
+    }
+    break;
+default:
+    if (mAcch.ChkGroundLanding()) {
+       // stuff 1
+    }
+    break;
+}
+```
+You can also get single case switch statements instead of `if/else`. For example,
+```c++
+if (m_itemNo == BOKO_BELT) {
+    offsetY = -24.0f;
+}
+// Should be
+switch (m_itemNo) {
+case BOKO_BELT:
+    offsetY = -24.0f;
+    break;
+}
+```
+The above is especially useful to try out in case you're dealing with register mismatch.
+The assembly will hint at as the switch statement if you get this pattern,
+```asm
+cmpwi
+beq
+bge
+;
+cmpwi
+beq
+b
+;; Aka cmpwi + a lot of branching is likely a small switch statement.
+```
+
+## Register Problems
+When dealing with register mismatching, we've found the following situations to help.
+When dealing with enums, you might have to do a double return. For example
+```c++
+return createInit() ? 4 : 5 
+// Matches, but we know Enums were used in this function.
+if (createInit()) {
+   return cPhs_COMPLEATE_e;
+} else {
+   return cPhs_ERROR_e;
+}
+```
+This pattern also holds for general `if` statements.
+```c++
+s32 modelFileName = (mArrowType == TYPE_LIGHT ? 0x38 : 0x37) & 0xFFFF;
+// Matches, but Enums are preferred.
+u16 modelFileIndex;
+if (mArrowType == TYPE_LIGHT) {
+    modelFileIndex = LINK_BOL_ARROWGLITTER;
+} else {
+    modelFileIndex = LINK_BOL_ARROW;
+}
+```
+
+## Struct Size
+Not all the Structs in Ghidra have been defined. You'll see array access syntax for undersized structs. Cross-reference 
+with [this resource](https://github.com/LagoLunatic/WW-Hacking-Docs/blob/master/Extracted%20Data/Actor%20Instance%20Sizes.txt)
+and adjust accordingly.
+
+As there are still plenty of fields we don't have names for, defining our struct is very useful, you'll see the following
+patterns in the codebase.
+
+```c++
+public:
+  /* 0xOFFSET */ type mName; 
+// Offset value for the field, Ghidra is useful for this.
+  /* 0xOFFSET */ type* mpName;
+// AKA standard C naming conventions.
+
+// The following are more "personal preference" approaches to handling unknown values.
+  /* 0xOFFSET */ u8 mOffset[0xHigh_Unknown - 0xOffset];
+// All the values between 0xHigh_Unknown - 0xOffset are still unknown.
+  /* 0xOFFSET  */ u8 field_0xOFFSET;
+  /* 0xOFFSET1 */ u8 field_0xOFFSET1;
+// Similar to the array approach, but defines all the unkown `u8` as their own field instead.
+  /* 0xOFFSET */  type field_0xOFFSET;
+// This field is typed correctly, but has an unknown purpose. `type mOffset` could also convey this. 
+} // Size 0xVALUE
+// ^ Total size of the Struct, could also do the folling
+STATIC_ASSERT(sizeof(MyStruct) == 0xVALUE);
+```
```

Comment:
this whole code block is really hard to read, I feel like it would be easier for someone to just look at an example of actual code instead of pseudocode, or better yet maybe mention the ghidra script that generates the struct in this format automatically so you don't have to type it out manually